src/v/cluster/tm_stm_cache.h

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/persisted_stm.h"
+#include "config/configuration.h"
+#include "features/feature_table.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+#include "model/timestamp.h"
+#include "raft/consensus.h"
+#include "raft/errc.h"
+#include "raft/logger.h"
+#include "raft/state_machine.h"
+#include "raft/types.h"
+#include "storage/snapshot.h"
+#include "utils/expiring_promise.h"
+#include "utils/mutex.h"
+
+#include <absl/container/btree_set.h>
+#include <absl/container/flat_hash_map.h>
+
+#include <compare>
+
+namespace cluster {
+
+// Update this if your patch bumps the version.
+// Current version of transaction record (v2).
+// Includes all changes for transactions GA.
+// + last_pid field - KIP-360 support
+// + transferring - tm_stm graceful failover support
+// + partition::topic_revision - revision_id of the topic
+struct tm_transaction {
+    static constexpr uint8_t version = 2;
```

Comment:
Did not understand this commit/ refactor, this needs a commit message, it looks like it is doing more than just moving the code around.